sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -555,6 +575,38 @@ public BundleFinalizer getBundleFinalizer() {
     }
   }
 
+  private void embedOutboundElementsIfApplicable(
+      ProcessBundleResponse.Builder response, BundleProcessor bundleProcessor) {
+    List<Elements> collectedElements =
+        new ArrayList<>(bundleProcessor.getOutboundAggregators().size());
+    boolean hasFlushedAggregator = false;
+    for (BeamFnDataOutboundAggregator aggregator :
```

Comment:
Add a comment by the instantiation of the outbound aggregator map that your using a LinkedHashMap since you rely on stable iteration order.